packages/dds/tree/src/feature-libraries/editable-tree/editableTreeContext.ts

```diff
@@ -91,6 +99,39 @@ export interface EditableTreeContext {
      * is committed successfully.
      */
     attachAfterChangeHandler(afterChangeHandler: (context: EditableTreeContext) => void): void;
+
+    /**
+     * Types an arbitrary data with the given `TreeSchemaIdentifier` and wraps it into a format,
+     * which can be accepted by the EditableTree when using simple assignments.
+     *
+     * Use it whenever the field is polymorphic, to explicitly define the type of your data.
+     */
+    newDetachedNode<T extends Brand<any, string>>(
+        type: TreeSchemaIdentifier,
+        data: T extends BrandedType<infer ValueType, infer Name>
+            ? BrandedType<ValueType, Name>
+            : never,
+    ): T;
+    /**
+     * Types an arbitrary data with the given `TreeSchemaIdentifier` and wraps it into a format,
+     * which can be accepted by the EditableTree when using simple assignments.
+     *
+     * Use it whenever the field is polymorphic, to explicitly define the type of your data.
+     */
+    newDetachedNode<T extends Brand<any, string>>(
+        type: TreeSchemaIdentifier,
+        data: T extends BrandedType<infer ValueType, string> ? ValueType : never,
+    ): T;
+    /**
+     * _Type-unsafe overload of the method allowing to change EditableTrees using simple assignments in contexts,
+     * where proper typing is not possible or not important._
+     *
+     * Types an arbitrary data with the given `TreeSchemaIdentifier` and wraps it into a format,
+     * which can be accepted by the EditableTree when using simple assignments.
+     *
+     * Use it whenever the field is polymorphic, to explicitly define the type of your data.
+     */
+    newDetachedNode<T extends Brand<any, string>>(type: TreeSchemaIdentifier, data: unknown): T;
```

Comment:
This is the most tricky part for me. It allows to use a very strict typing as well as no typing at all. However, if one e.g. misses to properly type a field in the object, the whole type check will be discarded i.e. the least type-safe overload will be used making everything meaningless. So, not sure at all, how to deal with it.
There are a lot of examples in `editableTree.editing.spec.ts` as well as in `mockData.ts`: check `getPerson` as an explicit example on how one could create such "detached node". `getHandles` there is another example of mapping types to the corresponding schemas, that was also demanded as probably a more convenient way of creating new typed nodes, with which one can avoid using `brand` everywhere instead.

Reply:
Our type branding library is not designed to be used this way. It's really designed just be used to define nominally typed type aliases, not for generic usage like this. I'll see if I can figure out a better approach.

Also, when dealing with type safe and non-type safe APIs, you almost always want to give them different names. Generally, the unsafe one would indicate in its name that its unsafe so people reading the code or calling it know they need to be extra careful to ensure the types are correct.: we want to ensure people looking at the code can tell if the compiler is checking the types of if they have to. (This is a similar motivation to why many languages have explicit "unsafe" blocks (like rust and C#)).

Reply:
I have made https://github.com/sharptrip/FluidFramework/pull/19 which I think does a good job of separating the schema aware and generic API, avoiding this complexity.

Reply:
Thanks a lot. Merged, cleaned, tested.